src/utils/device/clientInformation.ts

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient } from "matrix-js-sdk/src/client";
+
+import BasePlatform from "../../BasePlatform";
+import { IConfigOptions } from "../../IConfigOptions";
+
+const formatUrl = (): string | undefined => {
+    // don't record url for electron clients
+    if (window.electron) {
+        return undefined;
+    }
+
+    // strip query-string and fragment from uri
+    const url = new URL(window.location.href);
+
+    return [
+        url.host,
+        url.pathname.replace(/\/$/, ""), // Remove trailing slash if present
+    ].join("");
+};
+
+const getClientInformationEventType = (deviceId: string): string =>
+    `io.element.matrix_client_information.${deviceId}`;
+
+/**
+ * Record extra client information for the current device
+ * https://github.com/vector-im/element-meta/blob/develop/spec/matrix_client_information.md
```

Comment:
https://github.com/vector-im/element-meta/blob/develop/spec/matrix_client_information.md#security-considerations are we sure there's no security considerations? This exposes additional metadata to your homeserver admin. Couldn't this live in SSSS?

Reply:
Hmm, EW already saves this information (stringified) as the device name, so it's not additional exactly. But it does remove the ability to easily overwrite it.
Using SSSS would limit this to only ever recording client information on verified devices.

Reply:
> Hmm, EW already saves this information (stringified) as the device name, so it's not additional exactly.

https://github.com/vector-im/element-meta/issues/382 wants to get rid of this

Reply:
That concern is more specifically about the a) user agent detail rather than the client version and b) exposing it to other users in device name. 
I'll update the docs with the concerns, and put this behind the labs flag so storing the extra data is opt in.

Reply:
https://github.com/vector-im/element-meta/pull/661

Reply:
@t3chguy what do you think?

Reply:
SGTM, thanks for resolving